### PR TITLE
[format] Format prefix in option keys are processed in each format

### PR DESCRIPTION
**Pull request by tsreaper**
### Purpose

Currently, format prefix in option keys are removed before sending into the constructor of each format. However, most formats (orc, parquet) still recover these prefixes in their code, so it would be better to let the format itself deal with these prefixes.

Also, we're going to introduce a special `compacted changelog` format, which wraps other file formats. However with the current code, each format can only see the options related to itself. If option keys are processed in each format, we can see options for other formats and keep what we need.

This PR process option keys in each format.

### Tests

Existing tests should cover this change.

### API and Format

No format changes.

### Documentation

No new feature.

### CODE REVIEW

File: paimon-format/src/main/java/org/apache/paimon/format/avro/AvroFileFormat.java [hunk 1]
```diff
@@ -62,11 +62,14 @@ public class AvroFileFormat extends FileFormat {
...
 
-    private final FormatContext context;
+    private final Options options;
+    private final int zstdLevel;
 
     public AvroFileFormat(FormatContext context) {
         super(IDENTIFIER);
-        this.context = context;
+
+        this.options = getIdentifierPrefixOptions(context.options(), false);
```
Comment (JingsongLi) [inline paimon-format/src/main/java/org/apache/paimon/format/avro/AvroFileFormat.java:71]: Can you also modify `AVRO_OUTPUT_CODEC` and `AVRO_ROW_NAME_MAPPING` to add prefix?

File: paimon-common/src/main/java/org/apache/paimon/format/FileFormat.java [hunk 2]
```diff
@@ -103,14 +110,19 @@ private static Optional<FileFormat> fromIdentifier(
...
-                        options.get(CoreOptions.WRITE_BATCH_SIZE),
-                        options.get(CoreOptions.FILE_COMPRESSION_ZSTD_LEVEL),
-                        options.get(CoreOptions.FILE_BLOCK_SIZE));
-        return FileFormat.fromIdentifier(formatIdentifier, context);
+    protected Options getIdentifierPrefixOptions(Options options, boolean keepPrefix) {
+        Map<String, String> result = new HashMap<>();
+        String prefix = formatIdentifier.toLowerCase() + ".";
+        for (String key : options.keySet()) {
+            if (key.toLowerCase().startsWith(prefix)) {
+                String substr = key.substring(prefix.length());
```
Comment (wwj6591812) [inline paimon-common/src/main/java/org/apache/paimon/format/FileFormat.java:118]: Change substr to another better name?

File: paimon-format/src/main/java/org/apache/paimon/format/avro/AvroFileFormat.java [hunk 3]
```diff
@@ -62,11 +62,14 @@ public class AvroFileFormat extends FileFormat {
     public static final ConfigOption<Map<String, String>> AVRO_ROW_NAME_MAPPING =
```
Comment (wwj6591812) [inline paimon-format/src/main/java/org/apache/paimon/format/avro/AvroFileFormat.java:62]: In AvroFileFormat, IDENTIFIER && AVRO_OUTPUT_CODEC && AVRO_ROW_NAME_MAPPING should be private.

File: paimon-common/src/main/java/org/apache/paimon/format/FileFormat.java [hunk 4]
```diff
@@ -74,9 +75,15 @@ public Optional<SimpleStatsExtractor> createStatsExtractor(
         return Optional.empty();
     }
 
-    @VisibleForTesting
     public static FileFormat fromIdentifier(String identifier, Options options) {
-        return fromIdentifier(identifier, new FormatContext(options, 1024, 1024));
+        return fromIdentifier(
```
Comment (wwj6591812) [inline paimon-common/src/main/java/org/apache/paimon/format/FileFormat.java:79]: This called method fromIdentifier should be private.

**Comment by wwj6591812** [comment]
Good work!
After download your code in my computer, I have found no big problems.
Already add three small suggest.

File: paimon-format/src/test/java/org/apache/paimon/format/orc/OrcFileFormatTest.java [hunk 5]
```diff
@@ -47,8 +47,8 @@ public void testAbsent() {
     @Test
     public void testPresent() {
         Options options = new Options();
-        options.setString("haha", "1");
-        options.setString("compress", "zlib");
+        options.setString("orc.haha", "1");
```
Comment (wwj6591812) [inline paimon-format/src/test/java/org/apache/paimon/format/orc/OrcFileFormatTest.java:50]: Change haha to other normal orc option?

**Comment by wwj6591812** [comment]
Good work!
After download your code in my computer, I have found no big problems.
Already add four small suggests.

**Comment by JingsongLi** [comment]
+1

**Pull request merged**